pkg/cmd/experimental/syncgroups/groupsyncer_test.go

```diff
@@ -77,54 +170,94 @@ var Group1Members []*ldap.Entry = []*ldap.Entry{Member1, Member2}
 var Group2Members []*ldap.Entry = []*ldap.Entry{Member2, Member3}
 var Group3Members []*ldap.Entry = []*ldap.Entry{Member3, Member4}
 
-// TestSync ensures that data is exchanged and rearranged correctly during the sync process.
-func TestSync(t *testing.T) {
-	testGroupLister := TestGroupLister{
-		GroupUIDs: []string{Group1UID, Group2UID, Group3UID},
+// TestGoodSync ensures that data is exchanged and rearranged correctly during the sync process.
+func TestGoodSync(t *testing.T) {
+	testGroupSyncer, tc := newTestSyncer()
+	_, errs := testGroupSyncer.Sync()
+	for _, err := range errs {
+		t.Errorf("unexpected sync error: %v", err)
 	}
-	testGroupMemberExtractor := TestGroupMemberExtractor{
-		MemberMapping: map[string][]*ldap.Entry{
-			Group1UID: Group1Members,
-			Group2UID: Group2Members,
-			Group3UID: Group3Members,
-		},
+
+	checkClientForGroups(tc, newDefaultOpenShiftGroups(testGroupSyncer.Host), t)
+}
+
+func TestListFails(t *testing.T) {
```

Comment:
> Why is this a valuable test?

It makes sure we exit in a controlled way as expected and don't progress the job any further on failed lists.